### PR TITLE
[SDK/CLI] Support flex flow local to cloud

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

- Support flex flow upload run to cloud.
- Skip prompty flow run upload
- Refine the upload test cases.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py [hunk 1]
```diff
@@ -172,12 +173,15 @@ async def _upload_snapshot(self) -> str:
         """Upload run snapshot to cloud."""
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py:173]: add some doc on the expected output

File: src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py [hunk 2]
```diff
@@ -172,12 +173,15 @@ async def _upload_snapshot(self) -> str:
...
         logger.debug(f"Uploading snapshot for run {self.run.name!r}.")
         local_folder = self.run_output_path / LocalStorageFilenames.SNAPSHOT_FOLDER
+        # for some types of flex flow, even there is no flow.flex.yaml in the original flow folder,
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py:176]: will this work for prompty? cc @lalala123123

Comment (0mza987) [reply src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py:180]: Skipped upload run for prompty run for now as offline synced.

File: src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py (hunk 1 above)
Comment (0mza987) [reply src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py]: Updated

**Review by wangchao1230 — DISMISSED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**